### PR TITLE
Set up a channel for sending vehicles to client

### PR DESCRIPTION
**Pull request by arkadyan**
This is purely the server-side work pulled out from the work on [build UI for displaying/updating a table of vehicles and their locations](https://app.asana.com/0/1112935048846093/1114820119722045)

### CODE REVIEW

**Comment by arkadyan** [comment]
@skyqrose is it ok with you if I pull this server-side work out early? I don't think there's really a risk of serious merge conflicts with the work on doing adding vehicle data, but it seems like it could be nice to see the entire picture at once.

File: lib/skate_web/channels/vehicles_channel.ex [hunk 1]
```diff
@@ -0,0 +1,18 @@
+defmodule SkateWeb.VehiclesChannel do
+  use SkateWeb, :channel
+
+  def join("vehicles:" <> route_id, _message, socket) do
+    socket = assign(socket, :route_id, route_id)
```
Comment (paulswartz) [inline lib/skate_web/channels/vehicles_channel.ex]: Do you need the route_id? I don't see it used elsewhere.

**Comment by arkadyan** [comment]
Adding tests for `VehiclesChannel` now.

File: lib/skate_web/channels/vehicles_channel.ex (hunk 1 above)
Comment (arkadyan) [reply lib/skate_web/channels/vehicles_channel.ex]: Nope, doesn't look like it.

**Comment by arkadyan** [comment]
Added tests

File: lib/skate_web/channels/vehicles_channel.ex [hunk 2]
```diff
@@ -1,8 +1,12 @@
 defmodule SkateWeb.VehiclesChannel do
   use SkateWeb, :channel
 
+  alias Phoenix.Socket
+  alias Realtime.{Server, Vehicle}
+
+  @spec join(topic :: String.t(), payload :: map, socket :: Socket.t()) ::
```
Comment (paulswartz) [inline lib/skate_web/channels/vehicles_channel.ex]: These already have types: https://github.com/phoenixframework/phoenix/blob/v1.4.3/lib/phoenix/channel.ex#L288

So you can simplify to `@impl Phoenix.Channel`

Comment (arkadyan) [reply lib/skate_web/channels/vehicles_channel.ex]: Great, thanks, I didn't know that.

Comment (paulswartz) [reply lib/skate_web/channels/vehicles_channel.ex]: Anything that has a behaviour works this way: GenServer, GenStage, and Plug are some other examples off the top of my head.

**Pull request merged**